### ISSUE TITLE
Getting error with missing open_timeout variable

### ISSUE THREAD

**Issue by awchiu23** [racinette/websockets_proxy#3]
Hi racinette,

I really appreciate your work in enabling proxy support for websockets.  I tried it today with the following installed:

websockets 12.0
websockets_proxy 0.1.0
python-socks 2.4.4

While testing, I'm getting the following error:

File "...\Lib\site-packages\websockets\legacy\client.py", line 646, in __await_impl_timeout__
    async with asyncio_timeout(self.open_timeout):
                               ^^^^^^^^^^^^^^^^^
AttributeError: 'ProxyConnect' object has no attribute 'open_timeout'

I'm following your examples when trying.  Maybe I'm doing something wrong?

Any advise is appreciated, thanks.

Andrew

**Comment by awchiu23** [comment]
It seems like I cannot Send() using your websocket object .

    await ws.send(json.dumps({'time':int(time.time()), 'channel':channel, 'event':'subscribe', 'payload':payload, 'auth':auth}))
          ^^^^^^^
AttributeError: 'ProxyConnect' object has no attribute 'send'

Is it possible to send message through your object?

Thanks!

**Comment by racinette** [comment]
`proxy_connect` returns the same object as the original `websockets.connect`, so you must be using it wrong. You should post a minimal reproducible example of your code, so that I can run it and test it. Otherwise, it's just guessing.

**Comment by awchiu23** [comment]
Hi,

I have more info about the issue.   If I follow your example strictly as
such:

async with proxy_connect(proxy=XXX) as ws:
      ws.send(...)

It works.

But I wanted to return the websocket object from a function.  If I do this:

async def GetWebsocket():
     return proxy_connect(proxy=XXX)

ws = await GetWebsocket()
ws.send(...)

Then I would get error on the send() line:

AttributeError: 'ProxyConnect' object has no attribute 'send'

**Comment by racinette** [comment]
If you *really* want to return it from function, use it like so:

```
def GetWebsocket():
    return proxy_connect(proxy=XXX)

async def main():
    async with GetWebsocket() as ws:
         await ws.send("blablabla")
```

It should work.

Do not ask questions here, which have nothing to do with the library. It's not a library issue, it's a skill one. StackOverflow should be the way to go.

**Comment by awchiu23** [comment]
It works.  Thanks for all your help and advice.  I appreciate it!

Andrew

**Comment by yanghoxom** [comment]
@racinette how can I run it with timeout?
I want to increase the timeout of `connect`
because sometimes, endpoint become lag and it will cost few minutes to connect
look like default timeout is 30-40second

**Comment by racinette** [comment]
@yanghoxom you should try setting `proxy_conn_timeout` argument, as per docs.

In your example you set timeout to `4*60`, which is 4 minutes, however, you do not want the timeout to be long. 

I do not seem to understand the issue. I suppose setting the timeout to somewhere along the 10 seconds mark should do the trick.

**Comment by racinette** [comment]
@yanghoxom fixed in `0.1.2`
